applications/RANSApplication/custom_conditions/incompressible_potential_flow/incompressible_potential_flow_pressure_body_force_condition.h

```diff
@@ -0,0 +1,286 @@
+      */
+    Condition::Pointer Create(
+        IndexType NewId,
+        NodesArrayType const& ThisNodes,
+        PropertiesType::Pointer pProperties) const override
+    {
+        return Kratos::make_intrusive<IncompressiblePotentialFlowPressureBodyForceCondition>(
+            NewId, this->GetGeometry().Create(ThisNodes), pProperties);
+    }
+
+    Condition::Pointer Create(
+        IndexType NewId,
+        GeometryType::Pointer pGeom,
+        PropertiesType::Pointer pProperties) const override
+    {
+        return Kratos::make_intrusive<IncompressiblePotentialFlowPressureBodyForceCondition>(
+            NewId, pGeom, pProperties);
+    }
+
+    /**
+     * Clones the selected element variables, creating a new one
+     * @param NewId the ID of the new element
+     * @param ThisNodes the nodes of the new element
+     * @param pProperties the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    Condition::Pointer Clone(
+        IndexType NewId,
+        NodesArrayType const& rThisNodes) const override
+    {
+        Condition::Pointer p_new_condition = Create(
+            NewId, this->GetGeometry().Create(rThisNodes), this->pGetProperties());
+
+        p_new_condition->SetData(this->GetData());
+        p_new_condition->SetFlags(this->GetFlags());
+
+        return p_new_condition;
+    }
+
+    int Check(const ProcessInfo& rCurrentProcessInfo) override;
+
+    void Initialize() override;
+
+    void EquationIdVector(
+        EquationIdVectorType& rResult,
+        const ProcessInfo& rCurrentProcessInfo) const override;
+
+    /// Returns a list of the element's Dofs
+    /**
+     * @param ElementalDofList the list of DOFs
+     * @param rCurrentProcessInfo the current process info instance
+     */
+    void GetDofList(
+        DofsVectorType& ConditionDofList,
+        const ProcessInfo& CurrentProcessInfo) const override;
+
+    void GetValuesVector(
+        VectorType& rValues,
+        int Step = 0) const override;
+
+    GeometryData::IntegrationMethod GetIntegrationMethod() override;
+
+    void CalculateLocalSystem(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLeftHandSide(
+        MatrixType& rLeftHandSideMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateRightHandSide(
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override;
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override;
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    double mNormalLength;
```

Comment:
Why do you save this? Aren't you computing the norm each time the `NORMAL` is called?

Reply:
removed this condition. Apparently, I didnt notice, it was not doing it suppose to do. I made this as an initialization process only.